packages/promotion/src/models/promotion-rule-value.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  BeforeCreate,
+  Entity,
+  ManyToOne,
+  OnInit,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+
+import { generateEntityId } from "@medusajs/utils"
+import PromotionRule from "./promotion-rule"
+
+@Entity()
+export default class PromotionRuleValue {
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @ManyToOne(() => PromotionRule, {
+    onDelete: "cascade",
+    fieldName: "promotion_rule_id",
+    index: "IDX_promotion_rule_promotion_rule_value_id",
+  })
+  promotion_rule: PromotionRule
+
+  @Property({ columnType: "text" })
+  value: string
+
+  @BeforeCreate()
+  onCreate() {
+    this.id = generateEntityId(this.id, "promo-rule-value")
```

Comment:
todo: We shouldn't mix camel and snake casing in prefixes. And we should generally try to condense these to a point of avoiding the casing entirely.

I suggest `prorulval`

Reply:
I think the idea around the id looks nice, it would prevent unnecessary conflict with other modules indeed and make it shorter 💪